public/app/features/plugins/sql/components/QueryEditor.tsx

```diff
@@ -13,9 +13,19 @@ import { QueryHeader } from './QueryHeader';
 import { RawEditor } from './query-editor-raw/RawEditor';
 import { VisualEditor } from './visual-query-builder/VisualEditor';
 
-type Props = QueryEditorProps<SqlDatasource, SQLQuery, SQLOptions>;
-
-export function SqlQueryEditor({ datasource, query, onChange, onRunQuery, range }: Props) {
+type Props<TSQLQuery extends SQLQuery, TSQLOptions extends SQLOptions> = QueryEditorProps<
```

Comment:
I don't think we need the generic "TSQLQuery extends SQLQuery" anywhere (since the ui for queries is the same), though making SQLOptions generic is probably needed since those settings differ between sql databases.

I think we just need:  https://github.com/grafana/grafana/blob/sql-angular-to-react/public/app/features/plugins/sql/datasource/SqlDatasource.ts#L30

Reply:
fixed